openfeature/exception.py

```diff
@@ -12,6 +14,24 @@ class ErrorCode(Enum):
     INVALID_CONTEXT = "INVALID_CONTEXT"
     GENERAL = "GENERAL"
 
+    @classmethod
+    def to_exception(
+        cls, error_code: ErrorCode, error_message: str
+    ) -> OpenFeatureError:
+        return typing.cast(
+            OpenFeatureError,
+            {
+                ErrorCode.PROVIDER_NOT_READY: ProviderNotReadyError,
+                ErrorCode.PROVIDER_FATAL: ProviderFatalError,
+                ErrorCode.FLAG_NOT_FOUND: FlagNotFoundError,
+                ErrorCode.PARSE_ERROR: ParseError,
+                ErrorCode.TYPE_MISMATCH: TypeMismatchError,
+                ErrorCode.TARGETING_KEY_MISSING: TargetingKeyMissingError,
+                ErrorCode.INVALID_CONTEXT: InvalidContextError,
+                ErrorCode.GENERAL: GeneralError,
+            }.get(error_code, GeneralError)(error_message),
```

Comment:
Yes, agree that pattern matching would be great. We can make it a class variable of the enum, which I like for cohesion, it just has a few quirks:

1) we need to move ErrorCode declaration to the bottom for the error constructors to be defined during module import
2) the keys are str's because they haven't been turned into enum instances yet by the Enum metaclass.
 
Check out the latest commit and let me know what you think.

Reply:
Nice, never thought about adding a class attribute to an enum 😄